ui/src/app/base/components/FormikFormContent/FormikFormContent.tsx

```diff
@@ -113,10 +114,10 @@ const FormikFormContent = <V,>({
   }, [initialValues, resetForm, resetOnSave, saved]);
 
   useEffect(() => {
-    if (!errors && hasSaved) {
-      onSuccess();
+    if (!hasErrors && hasSaved) {
+      onSuccess && onSuccess(values);
     }
-  }, [onSuccess, errors, hasSaved]);
+  }, [onSuccess, hasErrors, hasSaved, values]);
```

Comment:
Do you think there could be unintended side-effects if this `useEffect` runs every time `values` changes?

Reply:
It's going too run quite a lot, but I don't think that will be a problem? Was there anything you were thinking of? Or can you think of a better way to do this?

Reply:
I'd just been considering the case where the form doesn't close on success, i.e. the cleanup function doesn't get run to reset `saved`. In that scenario the `onSuccess` function will run every time values changes even though we only really want it to run when `hasSaved` changes from false to true.

I don't think it's an issue with any of our forms now (although maybe the "Save and add another" ones?) but it might be safer to only run the `onSuccess` function if `hasSaved` changes from false to true and not just when `hasSaved` is true. I guess one way of handling it would be with an internal `hasSaved` state that you explicitly set to false in the `useEffect`.

Reply:
Ah yeah I see. `hasSaved` does currently only get set when `saved` has cycled from `false` to `true`, but once it has cycled there's nothing to stop it from firing again, even if `saved` goes back to `false`.

How about when it calls `onSuccess` it sets a ref so that `onSuccess` can only run once. Forms that get reused (e.g. the save-and-add-another forms that you mentioned) call `resetOnSave` so when it resets the form it could also reset `onSuccessCalled`  and `hasSaved`?

Reply:
Yeah that sounds good!

Reply:
I've pushed that change. I've left it as a separate commit so you can just review the changes.

Reply:
LGTM